extra_data/tests/test_aliases.py

```diff
@@ -0,0 +1,294 @@
+        assert isinstance(kd1, KeyData)
+        assert isinstance(kd2, KeyData)
+        assert kd1.source == kd2.source
+        assert kd1.key == kd2.key
+        assert kd1.train_ids == kd2.train_ids
+
+    # Test whether source alias yields identical SourceData.
+    assert run.alias['sa3-xgm'] is run['SA3_XTD10_XGM/XGM/DOOCS']
+
+    # Test whether source alias plus literal key yields equal KeyData.
+    assert_equal_keydata(
+        run.alias['sa3-xgm', 'pulseEnergy.wavelengthUsed'],
+        run['SA3_XTD10_XGM/XGM/DOOCS', 'pulseEnergy.wavelengthUsed'])
+
+    # Test whether key alias yields equal KeyData.
+    assert_equal_keydata(
+        run.alias['hv'],
+        run['SA3_XTD10_XGM/XGM/DOOCS', 'pulseEnergy.wavelengthUsed'])
+
+    # Test undefined aliases.
+    with pytest.raises(AliasError):
+        run.alias['foo']
+        run.alias['foo', 'bar']
+
+    # Test using a literal key with a key alias.
+    with pytest.raises(ValueError):
+        run.alias['hv', 'pressure']
+
+    # Test using an existing source alias for a non-existing source.
+    with pytest.raises(SourceNameError):
+        run.alias['bogus-source']
+
+    # Test using an existing key alias for a non-existing key.
+    with pytest.raises(PropertyNameError):
+        run.alias['bogus-key']
+
+    # Test re-applying the same aliases.
+    run2 = run.with_aliases(mock_sa3_control_aliases)
+    assert run._aliases == run2._aliases
+
+    # Test adding additional aliases.
+    run3 = run.with_aliases({'foo': 'bar'})
+    assert set(run._aliases.keys()) < set(run3._aliases.keys())
+    assert 'foo' in run3._aliases
+
+    # Test adding conflicting aliases
+    with pytest.raises(ValueError):
+        run.with_aliases({'sa3-xgm': 'x'})
+
+    # Test dropping aliases again.
+    run4 = run.drop_aliases()
+    assert not run4._aliases
+
+
+def test_json_alias_file(mock_sa3_control_data, mock_sa3_control_aliases, tmp_path):
+    aliases_path = tmp_path / 'aliases.json'
+    aliases_path.write_text('''
+{
+    "sa3-xgm": "SA3_XTD10_XGM/XGM/DOOCS",
+    "SA3_XTD10_XGM/XGM/DOOCS": {
+        "hv": "pulseEnergy.wavelengthUsed",
+        "beam-x": "beamPosition.ixPos",
+        "beam-y": "beamPosition.iyPos"
+    },
+
+    "imgfel-frames": ["SA3_XTD10_IMGFEL/CAM/BEAMVIEW:daqOutput", "data.image.pixels"],
+    "imgfel-frames2": ["SA3_XTD10_IMGFEL/CAM/BEAMVIEW2:daqOutput", "data.image.pixels"],
+    "imgfel-screen-pos": ["SA3_XTD10_IMGFEL/MOTOR/SCREEN", "actualPosition"],
+    "imgfel-filter-pos": ["SA3_XTD10_IMGFEL/MOTOR/FILTER", "actualPosition"],
+
+    "mcp-adc": "SA3_XTD10_MCP/ADC/1",
+    "mcp-mpod": "SA3_XTD10_MCP/MCPS/MPOD",
+    "mcp-voltage": ["SA3_XTD10_MCP/MCPS/MPOD", "channels.U3.voltage"],
+    "mcp-trace": ["SA3_XTD10_MCP/ADC/1:channel_5.output", "data.rawData"],
+
+    "bogus-source": "SA4_XTD20_XGM/DOOCS/MAIN",
+    "bogus-key": ["SA1_XTD2_XGM/DOOCS/MAIN", "foo"]
+}
+    ''')
+
+    run = H5File(mock_sa3_control_data).with_aliases(mock_sa3_control_aliases)
```

Comment:
```suggestion
    run = H5File(mock_sa3_control_data).with_aliases(aliases_path)
```

I just noticed that codecov shows the file-reading code is not being exercised. Same for the other two tests.